docs/snippets/GPU_Metric1.cpp

```diff
@@ -1,25 +1,23 @@
-#include <ie_core.hpp>
+#include <openvino/runtime/core.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
 
 int main() {
-using namespace InferenceEngine;
 //! [part1]
-InferenceEngine::Core core;
-CNNNetwork cnnNetwork = core.ReadNetwork("network.xml");
+ov::Core core;
+std::shared_ptr<ov::Model> network = core.read_model("sample.xml");
 uint32_t n_streams = 2;
 int64_t available_device_mem_size = 3221225472;
+std::map<std::string, ov::Any> options = {{ov::hint::model_ptr.name(), network}}; // Required. Set the address of the target network. If this is not set, the MAX_BATCH_SIZE returns 1.
+options.insert(std::make_pair(ov::streams::num.name(), n_streams)); // Optional. Set only when you want to estimate max batch size for a specific throughtput streams. Default is 1 or throughtput streams set by SetConfig.
+options.insert(std::make_pair(ov::intel_gpu::hint::available_device_mem.name(), available_device_mem_size)); // Optional. Set only when you want to limit the available device mem size.
 
-std::map<std::string, Parameter> options = {{"MODEL_PTR", cnnNetwork.getFunction()}}; // Required. Set the address of the target network. If this is not set, the MAX_BATCH_SIZE returns 1.
-options.insert(std::make_pair("GPU_THROUGHPUT_STREAMS", n_streams)); // Optional. Set only when you want to estimate max batch size for a specific throughtput streams. Default is 1 or throughtput streams set by SetConfig.
-options.insert(std::make_pair("AVAILABLE_DEVICE_MEM_SIZE", available_device_mem_size)); // Optional. Set only when you want to limit the available device mem size.
-
-auto max_batch_size = core.GetMetric("GPU", GPU_METRIC_KEY(MAX_BATCH_SIZE), options).as<uint32_t>();
+uint32_t max_batch_size = core.get_property("GPU", ov::max_batch_size.name(), options);
 //! [part1]
 //! [part2]
-std::map<std::string, Parameter> opt = {{"MODEL_PTR", cnnNetwork.getFunction()}}; // Required. Same usage as for the MAX_BATCH_SIZE above. If not set, the OPTIONAL_BATCH_SIZE returns 1.
 // This is not entirely GPU-specific metric (so METRIC_KEY is used rather than GPU_METRIC_KEY below),
 // but the GPU is the only device that supports that at the moment.
 // For the GPU, the metric already accommodates limitation for the on-device memory that the MAX_BATCH_SIZE poses.
 // so OPTIMAL_BATCH_SIZE is always less than MAX_BATCH_SIZE. Unlike the latter it is also aligned to the power of 2.
-auto optimal_batch_size = core.GetMetric("GPU", METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
+uint32_t optimal_batch_size = core.get_property("GPU", ov::optimal_batch_size.name(), options);
```

Comment:
why `ov::optimal_batch_size.name()` instead of `ov::optimal_batch_size`

Reply:
There is no implementation for `get_property(std::string, ov::Property, ov::AnyMap)` method
For now we have only two options with optional arguments for get_property method:
`get_property(const std::string& device_name, const std::string& name, const AnyMap& arguments)`
Or:
`get_property(const std::string& deviceName, const ov::Property<T, M>& property, Args&&... args)`

Reply:
In the latest PR https://github.com/openvinotoolkit/openvino/pull/9993/files/943141df0d24441f452ef7662245c3ad802498bd..bee150fea9ad7bb4f9360d2547cd5bedf2254cda we have added an ability to provide property instead of it's name. Can be fixed now